### PR TITLE
timezone aware

### PR DESCRIPTION
**Pull request by AlanKuurstra**
addresses #777 

Adds capability to handle datetime stripping for all datetimes allowed in the dicom and bids standards.

### CODE REVIEW

**Comment by yarikoptic** [comment]
dang, I have managed to miss this PR and CI managed to not run test for some reason!
I will attend to it within a few days.  If you could rebase or otherwise refresh it to re-trigger the CI, would be appreciated.

**Comment by AlanKuurstra** [comment]
I rebased, CI is awaiting approval from a maintainer.

File: heudiconv/utils.py
Comment (yarikoptic) [inline heudiconv/utils.py:766]: I am not sure who we can receive a datetime_obj with timezone from above, but I think it better be explicitly checked and warned about so we do not quietly overload it:

```suggestion
    if utc_offset_dcm := dcm_data.get([0x0008, 0x0201]):
        if utc_offset := utf_offset_dcm.value:
             datetime_obj2 = datetime_utc_offset(datetime_obj, utc_offset)
             if datetime_obj.tzinfo and datetime_obj2 != datetime_obj:
                 lgr.warning("Unexpectedly previously parsed datetime %s contains zoneinfo which is different from the one obtained from DICOMs UTFOffset field: %s", datetime_obj, datetime_obj2)
             else:
                 datetime_obj = datetime_obj2 
```

or alike?

File: heudiconv/utils.py
Comment (yarikoptic) [inline heudiconv/utils.py:671]: Please keep prior implementation around, but add a deprecation warning. It could be that someone used it in their heuristic, so we better do not just rip it from under them.

File: heudiconv/tests/test_utils.py
Comment (yarikoptic) [inline heudiconv/tests/test_utils.py:256]: It is unfortunate that we do not carry any "proper" dicom file with timezone info. But FWIW our dependencies do:

```
❯ /usr/bin/find -iname *.dcm | while read d; do off=$(dcmdump +P 0008,0201 $d 2>/dev/null) ; [ -n "$off" ] || continue; echo -e "$d\t$off"; done | head
./.tox/typing/lib/python3.11/site-packages/pydicom/data/charset_files/chrArab.dcm	(0008,0201) SH [-0400]                                  #   6, 1 TimezoneOffsetFromUTC
./.tox/typing/lib/python3.11/site-packages/pydicom/data/charset_files/chrFren.dcm	(0008,0201) SH [-0400]                                  #   6, 1 TimezoneOffsetFromUTC
./.tox/typing/lib/python3.11/site-packages/pydicom/data/charset_files/chrFrenMulti.dcm	(0008,0201) SH [-0400]                                  #   6, 1 TimezoneOffsetFromUTC

```

but I do not think we should rely on them.  But we could copy some I guess. WDYT, worth it?

File: heudiconv/dicoms.py
Comment (yarikoptic) [inline heudiconv/dicoms.py:536]: FWIW this is not functionally equivalent to above since we did `strip()` on values, so if string of spaces is a value for some reason (don't want to dig now why we even did such safe guarding), it would not be considered, which is the case here.  Could you adjust so we stay 100% consistent with prior implementation?

**Review by yarikoptic — CHANGES_REQUESTED** [review]
Thanks again for the PR! Let's get "zone aware!" ;-)
See individual comments for some minor tune ups requested

File: heudiconv/tests/test_utils.py
Comment (AlanKuurstra) [reply heudiconv/tests/test_utils.py:275]: I'm not sure I see the benefit to adding dicom files to the git repository. Generating the dicom from scratch facilitates testing every tag allowed by the standard without keeping numerous files in the repository.

Comment (yarikoptic) [reply heudiconv/tests/test_utils.py:276]: there are pros/cons for both approaches. e.g. for existing/committed (to git or via git-annex) files -- ensuring that we are operating correctly on the files as provided by manufacturers. Not surprising that dcm2niix has all those collections linked as submodules: https://github.com/rordenlab/dcm2niix/blob/master/.gitmodules

File: heudiconv/dicoms.py
Comment (yarikoptic) [reply heudiconv/dicoms.py:540]: not sure why it was marked resolved since  code remains the same -- if empty value we would proceed to the function and return

File: heudiconv/utils.py
Comment (yarikoptic) [inline heudiconv/utils.py:800]: please use `black` and/or ideally `pre-commit` which we have for the project -- it should take care about reformatting those long lines etc

File: heudiconv/utils.py
Comment (yarikoptic) [inline heudiconv/utils.py:779]: I know that you might have followed a trend of some docstrings sneaking in with `r` for no reason, but if no reason here in new ones -- please do not use them since adds cognitive burden to later possibly look for something needing this `raw` annotation

File: heudiconv/tests/test_utils.py
Comment (yarikoptic) [reply heudiconv/tests/test_utils.py:276]: but you are correct that on the fly generation could be quite handy for how you have done them above - in case of parametric tests etc. I was just wondering if we could add testing on some existing file with such metadata in it... but if none there -- let's forget about this particular aspect

**Review by yarikoptic — APPROVED** [review]
thank you @AlanKuurstra , let's proceed!

**Pull request merged**
